optuna/visualization/_optimization_history.py

```diff
@@ -23,6 +24,111 @@
+    studies: List[Study],
+    target: Optional[Callable[[FrozenTrial], float]],
+    target_name: str,
+) -> List[_OptimizationHistoryInfo]:
+    optimization_history_info_list = []
+    for study in studies:
+        trials = study.get_trials(states=(TrialState.COMPLETE,))
+        label_name = target_name if len(studies) == 1 else f"{target_name} of {study.study_name}"
+        best_values: Optional[List[float]]
+        best_label_name: Optional[str]
+        if target is None:
+            values = [cast(float, t.value) for t in trials]
+            if study.direction == StudyDirection.MINIMIZE:
+                best_values = list(np.minimum.accumulate(values))
+            else:
+                best_values = list(np.maximum.accumulate(values))
+            best_label_name = (
+                "Best Value" if len(studies) == 1 else f"Best Value of {study.study_name}"
+            )
+        else:
+            values = [target(t) for t in trials]
+            best_values = None
+            best_label_name = None
+        optimization_history_info_list.append(
+            _OptimizationHistoryInfo(
+                trial_numbers=[t.number for t in trials],
+                values=values,
+                label_name=label_name,
+                best_values=best_values,
+                best_label_name=best_label_name,
+            )
+        )
+    return optimization_history_info_list
+
+
+class _OptimizationHistoryErrorBarInfo(NamedTuple):
+    trial_numbers: List[int]
+    value_means: List[float]
+    value_stds: List[float]
+    label_name: str
+    best_value_means: Optional[List[float]]
+    best_value_stds: Optional[List[float]]
+    best_label_name: Optional[str]
+
+
+def _get_optimization_history_error_bar_info(
+    studies: List[Study],
+    target: Optional[Callable[[FrozenTrial], float]],
+    target_name: str,
+) -> _OptimizationHistoryErrorBarInfo:
+    info_list = _get_optimization_history_info_list(studies, target, target_name)
+    max_trial_number = max(max(info.trial_numbers) for info in info_list)
+
+    # Calculate mean and std of target values for each trial number.
+    values: List[List[float]] = [[] for _ in range(max_trial_number + 2)]
+    # TODO(knshnb): Took over `+2` from the previous code, but not sure why it is necessary.
+    for info in info_list:
+        for trial_number, value in zip(info.trial_numbers, info.values):
+            values[trial_number].append(value)
+    mean_of_values = [np.mean(v) if len(v) > 0 else None for v in values]
+    std_of_values = [np.std(v) if len(v) > 0 else None for v in values]
+    trial_numbers = np.arange(max_trial_number + 2)[[v is not None for v in mean_of_values]]
+    value_means = np.asarray(mean_of_values)[trial_numbers]
+    value_stds = np.asarray(std_of_values)[trial_numbers]
+
+    best_value_means: Optional[List[float]]
+    best_value_stds: Optional[List[float]]
+    best_label_name: Optional[str]
+    if target is None:
```

Comment:
I'm sorry, I didn't understand why you only calculate `best_values` when `target is None`. I thought that has nothing to do with `target`.